compaction/compaction_manager.cc

```diff
@@ -462,8 +467,7 @@ class sstables_task_executor : public compaction_task_executor, public sstables_
         set_sstables(std::move(sstables));
     }
 
-    virtual ~sstables_task_executor();
-
+    virtual void stop_compaction_executor() override;
```

Comment:
I'm not fancy about the "stop_" name since usually stop methods are async and wait for async cleanup of the object to complete.

How about `unregister()` or `release_resources()`?